infrastructure/via/src/bootstrap.ts

```diff
@@ -23,42 +24,141 @@ async function updateEnvVariable(envFilePath: string, variableName: string, newV
+        txids.push(data.propose_sequencer_tx_id);
+
+        // Process the Attestation
+        for (let i = 0; i < files.length; i++) {
+            const data = JSON.parse(await fs.readFile(path.join(genesisDir, files[i]), 'utf-8'));
+            if (data.tx_type == 'Attest') {
+                txids.push(data.tx_id);
+            }
+        }
+        genesisTxIds = txids.join(',');
+    }
 
     const envFilePath = path.join(process.env.VIA_HOME!, `etc/env/target/${process.env.VIA_ENV}.env`);
-
     console.log(`Updating file ${envFilePath}`);
 
-    await updateEnvVariable(envFilePath, 'VIA_BTC_WATCH_BOOTSTRAP_TXIDS', newTxids);
+    await updateEnvVariable(envFilePath, 'VIA_GENESIS_BOOTSTRAP_TXIDS', genesisTxIds);
 
-    console.log(`Updated VIA_BTC_WATCH_BOOTSTRAP_TXIDS with: ${newTxids}`);
+    console.log(`Updated VIA_GENESIS_BOOTSTRAP_TXIDS with: ${genesisTxIds}`);
+}
 
-    try {
-        // await fs.unlink(txidsFilePath);
-        console.log(`NOT Deleted txids.via file.`);
-    } catch (error) {
-        console.error(`Error deleting txids.via file`);
+export async function attestProposedSequencer(
+    network: string,
+    rpcUrl: string,
+    rpcUsername: string,
+    rpcPassword: string,
+    privateKey: string,
+    proposeSequencerFile: string
+) {
+    process.chdir(`${process.env.VIA_HOME}`);
+
+    if (!proposeSequencerFile) {
+        proposeSequencerFile = `etc/env/via/genesis/${network}/SystemBootstrapping.json`;
     }
+    const proposeSequencer = JSON.parse(await fs.readFile(proposeSequencerFile, 'utf-8'));
+
+    let cmd = `cargo run --example bootstrap ${network} ${rpcUrl} ${rpcUsername} ${rpcPassword} Attest ${privateKey} `;
+    cmd += `${proposeSequencer['propose_sequencer_tx_id']}`;
+
+    await utils.spawn(cmd);
 }
 
-export async function via_bootstrap(network: string, rpcUrl: string, rpcUsername: string, rpcPassword: string) {
+export async function systemBootstrapping(
+    network: string,
+    rpcUrl: string,
+    rpcUsername: string,
+    rpcPassword: string,
+    privateKey: string,
+    startBlock: string,
+    verifiersPubKeys: string,
+    governanceAddress: string,
+    bridgeAddress: string,
+    sequencerAddress: string
+) {
     process.chdir(`${process.env.VIA_HOME}`);
-    await utils.spawn(`cargo run --example bootstrap -- ${network} ${rpcUrl} ${rpcUsername} ${rpcPassword}`);
 
-    await updateBootstrapTxidsEnv();
+    const genesisPath = `etc/env/file_based/genesis.yaml`;
+    const file = await fs.readFile(genesisPath, 'utf-8');
+    const genesisData = parse(file);
+
+    const default_aa_hash = genesisData['default_aa_hash'];
+    const bootloader_hash = genesisData['bootloader_hash'];
+
+    let cmd = `cargo run --example bootstrap ${network} ${rpcUrl} ${rpcUsername} ${rpcPassword} SystemBootstrapping ${privateKey} `;
+    cmd += `${startBlock} ${verifiersPubKeys} ${default_aa_hash} ${bootloader_hash} ${governanceAddress} ${bridgeAddress} ${sequencerAddress}`;
```

Comment:
Since now we have a big amount of arguments passed to this, maybe we should consider switching to named arguments and using `clap` crate to handle them, what do you think?

Reply:
On second though, no need to change this, since we probably won't ever call the bootstrap example directly